### ISSUE TITLE
graphql-codegen-cli UnhandledPromiseRejection since 1.8.0

### ISSUE THREAD

**Issue by pristas-peter** [dotansimha/graphql-code-generator#2707]
**Describe the bug**
Since 1.8.0, when generator is run through code and there is a parsing error, unhandled rejection pops out and node process crashes. 

**To Reproduce**

Version 1.8.0
```
Welcome to Node.js v12.9.1.
Type ".help" for more information.
> 
> const { generate } = require('@graphql-codegen/cli');
undefined
> p = generate({schema: 'schema.graphql', documents: 'src/**/*.{ts,tsx}', generates: {'/tmp/types.ts': { plugins: ['typescript', 'typescript-operations'] }}}).catch(console.error)
Promise { <pending> }
  ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate /tmp/types.ts
      ✔ Load GraphQL schemas
      ⠼ Load GraphQL documents
        Generate
(node:67419) UnhandledPromiseRejectionWarning: GraphQLError: Syntax Error: Expected {, found }
    at syntaxError (/Users/peter/Developer/pjur/node_modules/graphql/error/syntaxError.js:15:10)
    at Parser.expectToken (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1404:40)
    at Parser.many (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1514:10)
    at Parser.parseSelectionSet (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:267:24)
    at Parser.parseFragment (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:375:26)
    at Parser.parseSelection (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:280:58)
    at Parser.many (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1518:26)
    at Parser.parseSelectionSet (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:267:24)
  ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate /tmp/types.ts
      ✔ Load GraphQL schemas
      ✖ Load GraphQL documents
        → Syntax Error: Expected {, found }
        Generate


 Found 1 error

  ✖ GraphQL request
    GraphQLError: Syntax Error: Expected {, found }
        at syntaxError (/Users/peter/Developer/pjur/node_modules/graphql/error/syntaxError.js:15:10)
        at Parser.expectToken (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1404:40)
        at Parser.many (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1514:10)
        at Parser.parseSelectionSet (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:267:24)
        at Parser.parseFragment (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:375:26)
        at Parser.parseSelection (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:280:58)
        at Parser.many (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1518:26)
        at Parser.parseSelectionSet (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:267:24)
        at Parser.parseField (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:308:68)
        at Parser.parseSelection (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:280:81)


ListrError: Something went wrong
    at /Users/peter/Developer/pjur/node_modules/listr/index.js:102:18
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at Object.executeCodegen (/Users/peter/Developer/pjur/node_modules/@graphql-codegen/cli/src/codegen.ts:275:3)
    at generate (/Users/peter/Developer/pjur/node_modules/@graphql-codegen/cli/src/generate-and-save.ts:74:23) {
  name: 'ListrError',
  errors: [
    GraphQLError: Syntax Error: Expected {, found }
        at syntaxError (/Users/peter/Developer/pjur/node_modules/graphql/error/syntaxError.js:15:10)
        at Parser.expectToken (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1404:40)
        at Parser.many (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1514:10)
        at Parser.parseSelectionSet (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:267:24)
        at Parser.parseFragment (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:375:26)
        at Parser.parseSelection (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:280:58)
        at Parser.many (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1518:26)
        at Parser.parseSelectionSet (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:267:24)
        at Parser.parseField (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:308:68)
        at Parser.parseSelection (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:280:81) {
      message: 'Syntax Error: Expected {, found }',
      locations: [Array],
      context: [Object: null prototype] {}
    }
  ],
  context: [Object: null prototype] {}
}
(node:67419) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 1)
```

Was correct in 1.7.0:

```
> p = generate({schema: 'schema.graphql', documents: 'src/**/*.{ts,tsx}', generates: {'/tmp/types.ts': { plugins: ['typescript', 'typescript-operations'] }}}).catch(console.error)
Promise { <pending> }
>   ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate /tmp/types.ts
      ✔ Load GraphQL schemas
      ✖ Load GraphQL documents
        → Syntax Error: Expected {, found }
        Generate


 Found 1 error

  ✖ GraphQL request
    Syntax Error: Expected {, found }

    GraphQL request:29:3
    28 |     ... on CheckoutFragment
    29 |   }
       |   ^
    30 | }


ListrError: Something went wrong
    at /Users/peter/Developer/pjur/node_modules/listr/index.js:102:18
    at Object.executeCodegen (/Users/peter/Developer/pjur/node_modules/@graphql-codegen/cli/src/codegen.ts:272:3)
    at generate (/Users/peter/Developer/pjur/node_modules/@graphql-codegen/cli/src/generate-and-save.ts:74:23) {
  name: 'ListrError',
  errors: [
    GraphQLError: Syntax Error: Expected {, found }
        at syntaxError (/Users/peter/Developer/pjur/node_modules/graphql/error/syntaxError.js:15:10)
        at Parser.expectToken (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1404:40)
        at Parser.many (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1514:10)
        at Parser.parseSelectionSet (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:267:24)
        at Parser.parseFragment (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:375:26)
        at Parser.parseSelection (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:280:58)
        at Parser.many (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:1518:26)
        at Parser.parseSelectionSet (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:267:24)
        at Parser.parseField (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:308:68)
        at Parser.parseSelection (/Users/peter/Developer/pjur/node_modules/graphql/language/parser.js:280:81) {
      message: 'Syntax Error: Expected {, found }',
      locations: [Array],
      context: [Object: null prototype] {}
    }
  ],
  context: [Object: null prototype] {}
}
```

1. My GraphQL schema:

```graphql
# Put your schema here
```

2. My GraphQL operations:

```graphql
# Put your operations here
```

3. My `codegen.yml` config file:

```yml
# Put your YML here
```

**Expected behavior**

**Environment:**

- OS: 
- `@graphql-codegen/...`: 
- NodeJS: 

**Additional context**

**Comment by dotansimha** [comment]
@pristas-peter Thanks, we'll take a look.

**Comment by dotansimha** [comment]
@pristas-peter we prefer to throw the error as-is. The `generate` method is `async` so you should add `await` when calling it, and if there is an error, it thrown. 

If you are not adding `await`, you'll get `UnhandledPromiseRejection` error because you are not handling this error correctly.

**Comment by pristas-peter** [comment]
Sorry that I havent mention that, of course that I wrap it it with try await catch. The problem still persist. Im using it in one of the gatsby lifecycle methods and since I havent control of how to catch that unhandled error (gatsby does it instead) the whole process exits during build.

**Comment by dotansimha** [comment]
I see. But still, the codegen works as async (and it was the same before, the only difference is that now we print the error and not only throw it, that's why you get this), and they only way to integrate it with other tools is to use `await` and catch the error. 

I can give it a try if you share a complete reproduction in a repo :)

**Comment by pristas-peter** [comment]
Will do

**Comment by pristas-peter** [comment]
@dotansimha https://github.com/pristas-peter/graphql-code-generator-2707

Just clone and run `npm i && npx gatsby develop`

**Comment by pristas-peter** [comment]
If you downgrade to 1.7 it works as expected

**Comment by dotansimha** [comment]
What do you mean by "works as expected"? you didn't get an error at all? or you just got the error in a different way?

**Comment by pristas-peter** [comment]
```
 try {
    await generate(
      {
        schema: "schema.graphql",
        documents: [
          "./node_modules/gatsby-transformer-*/**/*.js",
          "./src/!(graphql)/**/*.{ts,tsx}",
        ],
        config: { namingConvention: { enumValues: "keep" } },
        generates: {
          "./src/graphql/types.ts": {
            plugins: ["typescript", "typescript-operations"],
          },
          //   './src/graphql/definitions.ts': { plugins: ['codegen-graphql-to-gql'] },
          //   './src/graphql/fragment-types.json': { plugins: ['fragment-matcher'] },
        },
        pluckConfig: {
          modules: [
            {
              name: "gatsby",
              identifier: "graphql",
            },
          ],
        },
      },
      true
    )
  } catch (err) {
    console.error("not called")
  }
```
By expected I mean that the error handler in catch is called: `console.error("not called")` -> this would ouput to console not called and the build process won't exit and would continue.

**Comment by dotansimha** [comment]
Got it, thanks. I'll take a look!

**Comment by pristas-peter** [comment]
You can see the correct behaviour with:  
`npm i @graphql-codegen/cli@1.7.0 && npx gatsby develop`

Thank you :)

**Comment by ardatan** [comment]
Could you check with the following canary version?
 `1.8.2-alpha-ed9885c3.63+ed9885c3`

**Comment by pristas-peter** [comment]
Fixed. Thanks

**Comment by ardatan** [comment]
@pristas-peter Let's keep it open until we release new version :)

**Comment by dotansimha** [comment]
Fixed in 1.8.2.
